jni/include/knn_extension/faiss/utils/Heap.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <climits>
+#include <cmath>
+#include <cstring>
+
+#include <stdint.h>
+#include <cassert>
+#include <cstdio>
+
+#include <limits>
+#include <unordered_map>
+#include <faiss/utils/ordered_key_value.h>
+
+// Collection of heap operations with group id to dedupe
```

Comment:
group id is parent id correct? Could we make this consistent? 

Also, add more detail here about what ids are expected to be in the heap. Is it only expected to be parent ids?

Reply:
Will update. heap contains parent id during collection. At the end it converted to faiss id.